### ISSUE TITLE
Creating two windows-docker-machine instances destroys docker

### ISSUE THREAD

**Issue by jantimon** [StefanScherer/windows-docker-machine#67]
If you run two windows-docker-machine instances in parallel the second provisioning will destroy docker.

Any try to run docker (e.g. `docker --help`) will break with the following error:

```bash
parse "tcp://192.168.99.90 169.254.170.177:2376": invalid character " " in host name
```

The only way to repair docker is to remove the `currentContext` entry from `~/.docker/config.json`.

The reason is that windows-docker-machine will create the following docker context `meta.json`:

```json
{
  "Name": "2019-box",
  "Metadata": {
    "Description": "2019-box windows-docker-machine"
  },
  "Endpoints": {
    "docker": {
      "Host": "tcp://192.168.99.90 169.254.170.177:2376",
      "SkipTLSVerify": false
    }
  }
}
```

As you can see the `Endpoints.docker.Host` includes a space.  
My assumption is that the following code detects 2 ip addresses: https://github.com/StefanScherer/windows-docker-machine/blob/ae785e266905f12578c96fa6d4f1a73dd17f057b/scripts/create-machine.ps1#L11-L20

**Comment by StefanScherer** [comment]
Not only the space is wrong, there should be only one IP address.
Well, I normally run this in VMware environment, not in VirtualBox, so I don't have much experience how two boxes could work in parallel.

**Comment by jantimon** [comment]
I am not sure if it will work at all for two instances because of the ports.  
The main problem however is that docker breaks and it is really hard to find out why.

Do you have any idea how we might fix that spacing issues to prevent docker from breaking?

**Comment by StefanScherer** [comment]
Maybe after line 17 we have to get rid of everything after the space. Don‘t know why there are sometimes two IP addresses.

**Comment by jantimon** [comment]
Another solution would be to throw an error if two IP addresses are present.

But I would probably prefer to get rid of the space.

**Comment by StefanScherer** [comment]
I found a similar issue/pr https://github.com/StefanScherer/windows-docker-machine/pull/56 where we also couldn't find the reason.

But maybe it makes sense to filter out all IP addresses beginning with 169.254 🤔 

```
 ($_.IPAddress).StartsWith("169.254.")
```

and put it into the filterscript somehow.

**Comment by jantimon** [comment]
I like the idea 👍

<details>
  <summary>this is the full `Get-NetIPAddress` output I receive <b>(click to expand)</b></summary>

```
    2019-box: Caption                  : 
    2019-box: Description              : 
    2019-box: ElementName              : 
    2019-box: InstanceID               : 
    2019-box: CommunicationStatus      : 
    2019-box: DetailedStatus           : 
    2019-box: HealthState              : 
    2019-box: InstallDate              : 
    2019-box: Name                     : poB:DDC:noD>?o<D>opoDl<p=/;@55;@55;55;
    2019-box: OperatingStatus          : 
    2019-box: OperationalStatus        : 
    2019-box: PrimaryStatus            : 
    2019-box: Status                   : 
    2019-box: StatusDescriptions       : 
    2019-box: AvailableRequestedStates : 
    2019-box: EnabledDefault           : 2
    2019-box: EnabledState             : 
    2019-box: OtherEnabledState        : 
    2019-box: RequestedState           : 12
    2019-box: TimeOfLastStateChange    : 
    2019-box: TransitioningToState     : 12
    2019-box: CreationClassName        : 
    2019-box: SystemCreationClassName  : 
    2019-box: SystemName               : 
    2019-box: NameFormat               : 
    2019-box: OtherTypeDescription     : 
    2019-box: ProtocolIFType           : 4096
    2019-box: ProtocolType             : 
    2019-box: Address                  : 
    2019-box: AddressOrigin            : 0
    2019-box: AddressType              : 
    2019-box: IPv4Address              : 
    2019-box: IPv6Address              : fe80::90de:45e2:4efe:b2f3%16
    2019-box: IPVersionSupport         : 
    2019-box: PrefixLength             : 64
    2019-box: SubnetMask               : 
    2019-box: AddressFamily            : IPv6
    2019-box: AddressState             : Preferred
    2019-box: InterfaceAlias           : Ethernet 2
    2019-box: InterfaceIndex           : 16
    2019-box: IPAddress                : fe80::90de:45e2:4efe:b2f3%16
    2019-box: PreferredLifetime        : 10675199.02:48:05.4775807
    2019-box: PrefixOrigin             : WellKnown
    2019-box: SkipAsSource             : False
    2019-box: Store                    : ActiveStore
    2019-box: SuffixOrigin             : Link
    2019-box: Type                     : Unicast
    2019-box: ValidLifetime            : 10675199.02:48:05.4775807
    2019-box: PSComputerName           : 
    2019-box: ifIndex                  : 16
    2019-box: Caption                  : 
    2019-box: Description              : 
    2019-box: ElementName              : 
    2019-box: InstanceID               : 
    2019-box: CommunicationStatus      : 
    2019-box: DetailedStatus           : 
    2019-box: HealthState              : 
    2019-box: InstallDate              : 
    2019-box: Name                     : poB:DD>m<pDm<:pD@<BpDCo;@/;<55;<55;55;
    2019-box: OperatingStatus          : 
    2019-box: OperationalStatus        : 
    2019-box: PrimaryStatus            : 
    2019-box: Status                   : 
    2019-box: StatusDescriptions       : 
    2019-box: AvailableRequestedStates : 
    2019-box: EnabledDefault           : 2
    2019-box: EnabledState             : 
    2019-box: OtherEnabledState        : 
    2019-box: RequestedState           : 12
    2019-box: TimeOfLastStateChange    : 
    2019-box: TransitioningToState     : 12
    2019-box: CreationClassName        : 
    2019-box: SystemCreationClassName  : 
    2019-box: SystemName               : 
    2019-box: NameFormat               : 
    2019-box: OtherTypeDescription     : 
    2019-box: ProtocolIFType           : 4096
    2019-box: ProtocolType             : 
    2019-box: Address                  : 
    2019-box: AddressOrigin            : 0
    2019-box: AddressType              : 
    2019-box: IPv4Address              : 
    2019-box: IPv6Address              : fe80::4c2f:c20f:628f:9e16%12
    2019-box: IPVersionSupport         : 
    2019-box: PrefixLength             : 64
    2019-box: SubnetMask               : 
    2019-box: AddressFamily            : IPv6
    2019-box: AddressState             : Preferred
    2019-box: InterfaceAlias           : vEthernet (nat)
    2019-box: InterfaceIndex           : 12
    2019-box: IPAddress                : fe80::4c2f:c20f:628f:9e16%12
    2019-box: PreferredLifetime        : 10675199.02:48:05.4775807
    2019-box: PrefixOrigin             : WellKnown
    2019-box: SkipAsSource             : False
    2019-box: Store                    : ActiveStore
    2019-box: SuffixOrigin             : Link
    2019-box: Type                     : Unicast
    2019-box: ValidLifetime            : 10675199.02:48:05.4775807
    2019-box: PSComputerName           : 
    2019-box: ifIndex                  : 12
    2019-box: Caption                  : 
    2019-box: Description              : 
    2019-box: ElementName              : 
    2019-box: InstanceID               : 
    2019-box: CommunicationStatus      : 
    2019-box: DetailedStatus           : 
    2019-box: HealthState              : 
    2019-box: InstallDate              : 
    2019-box: Name                     : poB:DD?m@@DBCk=Dl?>@DBo>;/=55=55;55;
    2019-box: OperatingStatus          : 
    2019-box: OperationalStatus        : 
    2019-box: PrimaryStatus            : 
    2019-box: Status                   : 
    2019-box: StatusDescriptions       : 
    2019-box: AvailableRequestedStates : 
    2019-box: EnabledDefault           : 2
    2019-box: EnabledState             : 
    2019-box: OtherEnabledState        : 
    2019-box: RequestedState           : 12
    2019-box: TimeOfLastStateChange    : 
    2019-box: TransitioningToState     : 12
    2019-box: CreationClassName        : 
    2019-box: SystemCreationClassName  : 
    2019-box: SystemName               : 
    2019-box: NameFormat               : 
    2019-box: OtherTypeDescription     : 
    2019-box: ProtocolIFType           : 4096
    2019-box: ProtocolType             : 
    2019-box: Address                  : 
    2019-box: AddressOrigin            : 0
    2019-box: AddressType              : 
    2019-box: IPv4Address              : 
    2019-box: IPv6Address              : fe80::5c66:89a3:b546:8e41%3
    2019-box: IPVersionSupport         : 
    2019-box: PrefixLength             : 64
    2019-box: SubnetMask               : 
    2019-box: AddressFamily            : IPv6
    2019-box: AddressState             : Preferred
    2019-box: InterfaceAlias           : Ethernet
    2019-box: InterfaceIndex           : 3
    2019-box: IPAddress                : fe80::5c66:89a3:b546:8e41%3
    2019-box: PreferredLifetime        : 10675199.02:48:05.4775807
    2019-box: PrefixOrigin             : WellKnown
    2019-box: SkipAsSource             : False
    2019-box: Store                    : ActiveStore
    2019-box: SuffixOrigin             : Link
    2019-box: Type                     : Unicast
    2019-box: ValidLifetime            : 10675199.02:48:05.4775807
    2019-box: PSComputerName           : 
    2019-box: ifIndex                  : 3
    2019-box: Caption                  : 
    2019-box: Description              : 
    2019-box: ElementName              : 
    2019-box: InstanceID               : 
    2019-box: CommunicationStatus      : 
    2019-box: DetailedStatus           : 
    2019-box: HealthState              : 
    2019-box: InstallDate              : 
    2019-box: Name                     : DD;55;55;55;
    2019-box: OperatingStatus          : 
    2019-box: OperationalStatus        : 
    2019-box: PrimaryStatus            : 
    2019-box: Status                   : 
    2019-box: StatusDescriptions       : 
    2019-box: AvailableRequestedStates : 
    2019-box: EnabledDefault           : 2
    2019-box: EnabledState             : 
    2019-box: OtherEnabledState        : 
    2019-box: RequestedState           : 12
    2019-box: TimeOfLastStateChange    : 
    2019-box: TransitioningToState     : 12
    2019-box: CreationClassName        : 
    2019-box: SystemCreationClassName  : 
    2019-box: SystemName               : 
    2019-box: NameFormat               : 
    2019-box: OtherTypeDescription     : 
    2019-box: ProtocolIFType           : 4096
    2019-box: ProtocolType             : 
    2019-box: Address                  : 
    2019-box: AddressOrigin            : 0
    2019-box: AddressType              : 
    2019-box: IPv4Address              : 
    2019-box: IPv6Address              : ::1
    2019-box: IPVersionSupport         : 
    2019-box: PrefixLength             : 128
    2019-box: SubnetMask               : 
    2019-box: AddressFamily            : IPv6
    2019-box: AddressState             : Preferred
    2019-box: InterfaceAlias           : Loopback Pseudo-Interface 1
    2019-box: InterfaceIndex           : 1
    2019-box: IPAddress                : ::1
    2019-box: PreferredLifetime        : 10675199.02:48:05.4775807
    2019-box: PrefixOrigin             : WellKnown
    2019-box: SkipAsSource             : False
    2019-box: Store                    : ActiveStore
    2019-box: SuffixOrigin             : WellKnown
    2019-box: Type                     : Unicast
    2019-box: ValidLifetime            : 10675199.02:48:05.4775807
    2019-box: PSComputerName           : 
    2019-box: ifIndex                  : 1
    2019-box: Caption                  : 
    2019-box: Description              : 
    2019-box: ElementName              : 
    2019-box: InstanceID               : 
    2019-box: CommunicationStatus      : 
    2019-box: DetailedStatus           : 
    2019-box: HealthState              : 
    2019-box: InstallDate              : 
    2019-box: Name                     : ;C<8;@B8CC8C:55;@55;55;
    2019-box: OperatingStatus          : 
    2019-box: OperationalStatus        : 
    2019-box: PrimaryStatus            : 
    2019-box: Status                   : 
    2019-box: StatusDescriptions       : 
    2019-box: AvailableRequestedStates : 
    2019-box: EnabledDefault           : 2
    2019-box: EnabledState             : 
    2019-box: OtherEnabledState        : 
    2019-box: RequestedState           : 12
    2019-box: TimeOfLastStateChange    : 
    2019-box: TransitioningToState     : 12
    2019-box: CreationClassName        : 
    2019-box: SystemCreationClassName  : 
    2019-box: SystemName               : 
    2019-box: NameFormat               : 
    2019-box: OtherTypeDescription     : 
    2019-box: ProtocolIFType           : 4096
    2019-box: ProtocolType             : 
    2019-box: Address                  : 
    2019-box: AddressOrigin            : 0
    2019-box: AddressType              : 
    2019-box: IPv4Address              : 192.168.99.90
    2019-box: IPv6Address              : 
    2019-box: IPVersionSupport         : 
    2019-box: PrefixLength             : 24
    2019-box: SubnetMask               : 
    2019-box: AddressFamily            : IPv4
    2019-box: AddressState             : Duplicate
    2019-box: InterfaceAlias           : Ethernet 2
    2019-box: InterfaceIndex           : 16
    2019-box: IPAddress                : 192.168.99.90
    2019-box: PreferredLifetime        : 10675199.02:48:05.4775807
    2019-box: PrefixOrigin             : Manual
    2019-box: SkipAsSource             : False
    2019-box: Store                    : ActiveStore
    2019-box: SuffixOrigin             : Manual
    2019-box: Type                     : Unicast
    2019-box: ValidLifetime            : 10675199.02:48:05.4775807
    2019-box: PSComputerName           : 
    2019-box: ifIndex                  : 16
    2019-box: Caption                  : 
    2019-box: Description              : 
    2019-box: ElementName              : 
    2019-box: InstanceID               : 
    2019-box: CommunicationStatus      : 
    2019-box: DetailedStatus           : 
    2019-box: HealthState              : 
    2019-box: InstallDate              : 
    2019-box: Name                     : ;@C8<?>8;AB8<>=55;@55;55;
    2019-box: OperatingStatus          : 
    2019-box: OperationalStatus        : 
    2019-box: PrimaryStatus            : 
    2019-box: Status                   : 
    2019-box: StatusDescriptions       : 
    2019-box: AvailableRequestedStates : 
    2019-box: EnabledDefault           : 2
    2019-box: EnabledState             : 
    2019-box: OtherEnabledState        : 
    2019-box: RequestedState           : 12
    2019-box: TimeOfLastStateChange    : 
    2019-box: TransitioningToState     : 12
    2019-box: CreationClassName        : 
    2019-box: SystemCreationClassName  : 
    2019-box: SystemName               : 
    2019-box: NameFormat               : 
    2019-box: OtherTypeDescription     : 
    2019-box: ProtocolIFType           : 4096
    2019-box: ProtocolType             : 
    2019-box: Address                  : 
    2019-box: AddressOrigin            : 0
    2019-box: AddressType              : 
    2019-box: IPv4Address              : 169.254.178.243
    2019-box: IPv6Address              : 
    2019-box: IPVersionSupport         : 
    2019-box: PrefixLength             : 16
    2019-box: SubnetMask               : 
    2019-box: AddressFamily            : IPv4
    2019-box: AddressState             : Tentative
    2019-box: InterfaceAlias           : Ethernet 2
    2019-box: InterfaceIndex           : 16
    2019-box: IPAddress                : 169.254.178.243
    2019-box: PreferredLifetime        : 10675199.02:48:05.4775807
    2019-box: PrefixOrigin             : WellKnown
    2019-box: SkipAsSource             : False
    2019-box: Store                    : ActiveStore
    2019-box: SuffixOrigin             : Link
    2019-box: Type                     : Unicast
    2019-box: ValidLifetime            : 10675199.02:48:05.4775807
    2019-box: PSComputerName           : 
    2019-box: ifIndex                  : 16
    2019-box: Caption                  : 
    2019-box: Description              : 
    2019-box: ElementName              : 
    2019-box: InstanceID               : 
    2019-box: CommunicationStatus      : 
    2019-box: DetailedStatus           : 
    2019-box: HealthState              : 
    2019-box: InstallDate              : 
    2019-box: Name                     : ;C<8;@B8;@:8;55;<55;55;
    2019-box: OperatingStatus          : 
    2019-box: OperationalStatus        : 
    2019-box: PrimaryStatus            : 
    2019-box: Status                   : 
    2019-box: StatusDescriptions       : 
    2019-box: AvailableRequestedStates : 
    2019-box: EnabledDefault           : 2
    2019-box: EnabledState             : 
    2019-box: OtherEnabledState        : 
    2019-box: RequestedState           : 12
    2019-box: TimeOfLastStateChange    : 
    2019-box: TransitioningToState     : 12
    2019-box: CreationClassName        : 
    2019-box: SystemCreationClassName  : 
    2019-box: SystemName               : 
    2019-box: NameFormat               : 
    2019-box: OtherTypeDescription     : 
    2019-box: ProtocolIFType           : 4096
    2019-box: ProtocolType             : 
    2019-box: Address                  : 
    2019-box: AddressOrigin            : 0
    2019-box: AddressType              : 
    2019-box: IPv4Address              : 192.168.160.1
    2019-box: IPv6Address              : 
    2019-box: IPVersionSupport         : 
    2019-box: PrefixLength             : 20
    2019-box: SubnetMask               : 
    2019-box: AddressFamily            : IPv4
    2019-box: AddressState             : Preferred
    2019-box: InterfaceAlias           : vEthernet (nat)
    2019-box: InterfaceIndex           : 12
    2019-box: IPAddress                : 192.168.160.1
    2019-box: PreferredLifetime        : 10675199.02:48:05.4775807
    2019-box: PrefixOrigin             : Manual
    2019-box: SkipAsSource             : False
    2019-box: Store                    : ActiveStore
    2019-box: SuffixOrigin             : Manual
    2019-box: Type                     : Unicast
    2019-box: ValidLifetime            : 10675199.02:48:05.4775807
    2019-box: PSComputerName           : 
    2019-box: ifIndex                  : 12
    2019-box: Caption                  : 
    2019-box: Description              : 
    2019-box: ElementName              : 
    2019-box: InstanceID               : 
    2019-box: CommunicationStatus      : 
    2019-box: DetailedStatus           : 
    2019-box: HealthState              : 
    2019-box: InstallDate              : 
    2019-box: Name                     : ;:8:8<8;?55=55;55;
    2019-box: OperatingStatus          : 
    2019-box: OperationalStatus        : 
    2019-box: PrimaryStatus            : 
    2019-box: Status                   : 
    2019-box: StatusDescriptions       : 
    2019-box: AvailableRequestedStates : 
    2019-box: EnabledDefault           : 2
    2019-box: EnabledState             : 
    2019-box: OtherEnabledState        : 
    2019-box: RequestedState           : 12
    2019-box: TimeOfLastStateChange    : 
    2019-box: TransitioningToState     : 12
    2019-box: CreationClassName        : 
    2019-box: SystemCreationClassName  : 
    2019-box: SystemName               : 
    2019-box: NameFormat               : 
    2019-box: OtherTypeDescription     : 
    2019-box: ProtocolIFType           : 4096
    2019-box: ProtocolType             : 
    2019-box: Address                  : 
    2019-box: AddressOrigin            : 0
    2019-box: AddressType              : 
    2019-box: IPv4Address              : 10.0.2.15
    2019-box: IPv6Address              : 
    2019-box: IPVersionSupport         : 
    2019-box: PrefixLength             : 24
    2019-box: SubnetMask               : 
    2019-box: AddressFamily            : IPv4
    2019-box: AddressState             : Preferred
    2019-box: InterfaceAlias           : Ethernet
    2019-box: InterfaceIndex           : 3
    2019-box: IPAddress                : 10.0.2.15
    2019-box: PreferredLifetime        : 23:59:32
    2019-box: PrefixOrigin             : Dhcp
    2019-box: SkipAsSource             : False
    2019-box: Store                    : ActiveStore
    2019-box: SuffixOrigin             : Dhcp
    2019-box: Type                     : Unicast
    2019-box: ValidLifetime            : 23:59:32
    2019-box: PSComputerName           : 
    2019-box: ifIndex                  : 3
    2019-box: Caption                  : 
    2019-box: Description              : 
    2019-box: ElementName              : 
    2019-box: InstanceID               : 
    2019-box: CommunicationStatus      : 
    2019-box: DetailedStatus           : 
    2019-box: HealthState              : 
    2019-box: InstallDate              : 
    2019-box: Name                     : ;<A8:8:8;55;55;55;
    2019-box: OperatingStatus          : 
    2019-box: OperationalStatus        : 
    2019-box: PrimaryStatus            : 
    2019-box: Status                   : 
    2019-box: StatusDescriptions       : 
    2019-box: AvailableRequestedStates : 
    2019-box: EnabledDefault           : 2
    2019-box: EnabledState             : 
    2019-box: OtherEnabledState        : 
    2019-box: RequestedState           : 12
    2019-box: TimeOfLastStateChange    : 
    2019-box: TransitioningToState     : 12
    2019-box: CreationClassName        : 
    2019-box: SystemCreationClassName  : 
    2019-box: SystemName               : 
    2019-box: NameFormat               : 
    2019-box: OtherTypeDescription     : 
    2019-box: ProtocolIFType           : 4096
    2019-box: ProtocolType             : 
    2019-box: Address                  : 
    2019-box: AddressOrigin            : 0
    2019-box: AddressType              : 
    2019-box: IPv4Address              : 127.0.0.1
    2019-box: IPv6Address              : 
    2019-box: IPVersionSupport         : 
    2019-box: PrefixLength             : 8
    2019-box: SubnetMask               : 
    2019-box: AddressFamily            : IPv4
    2019-box: AddressState             : Preferred
    2019-box: InterfaceAlias           : Loopback Pseudo-Interface 1
    2019-box: InterfaceIndex           : 1
    2019-box: IPAddress                : 127.0.0.1
    2019-box: PreferredLifetime        : 10675199.02:48:05.4775807
    2019-box: PrefixOrigin             : WellKnown
    2019-box: SkipAsSource             : False
    2019-box: Store                    : ActiveStore
    2019-box: SuffixOrigin             : WellKnown
    2019-box: Type                     : Unicast
    2019-box: ValidLifetime            : 10675199.02:48:05.4775807
    2019-box: PSComputerName           : 
    2019-box: ifIndex                  : 1
```

</details>

I am not that familiar with powershell but I guess we would need the following statement:

```
$machineIp=(Get-NetIPAddress -AddressFamily IPv4 `
    | Where-Object -FilterScript { `
      ( ! ($_.InterfaceAlias).StartsWith("vEthernet (") ) `
      -And $_.IPAddress -Ne "127.0.0.1" `
      -And $_.IPAddress -Ne "10.0.2.15" `
      -And !($_.IPAddress).StartsWith("169.254.") `
    }).IPAddress
```

I'll give it a try

**Comment by StefanScherer** [comment]
Fixed with #69
